tests/RegisterTest.php

```diff
@@ -55,4 +77,256 @@ public function testSuccessfullRegister()
+    }
+
+    public function testUnsuccessfulRegisterTooShortUsername(): void
+    {
+        $data = [
+            'username' => 'Abc',
+            'password' => 'TestPassword_5',
+            'confirm_password' => 'TestPassword_5',
+            'email' => 'testemail5@test.mail',
+            'name' => 'TestName5',
+            'surname' => 'TestSurname5'
+        ];
+        $this->registerPostRequest($data);
+
+        $this->assertEquals(400, $this->client->getResponse()->getStatusCode());
+
+        $expectedErrorCode = '9ff3fdc4-b214-49db-8718-39c315e33d45';
+
+        $this->assertEquals('username', $this->getPropertyPath());
+        $this->assertEquals($expectedErrorCode, $this->getErrorCode());
+    }
+
+    public function testUnsuccessfulRegisterTooLongUsername(): void
+    {
+        $data = [
+            'username' => 'UsernameUsernameUsernameUsernameUsernameUsernameUsername',
+            'password' => 'TestPassword_6',
+            'confirm_password' => 'TestPassword_6',
+            'email' => 'testemail6@test.mail',
+            'name' => 'TestName6',
+            'surname' => 'TestSurname6'
+        ];
+        $this->registerPostRequest($data);
+
+        $this->assertEquals(400, $this->client->getResponse()->getStatusCode());
+
+        $expectedErrorCode = 'd94b19cc-114f-4f44-9cc4-4138e80a87b9';
+
+        $this->assertEquals('username', $this->getPropertyPath());
+        $this->assertEquals($expectedErrorCode, $this->getErrorCode());
+    }
+
+    public function testUnsuccessfulRegisterPasswordsDontMatch(): void
+    {
+        $data = [
+            'username' => 'TestUsername7',
+            'password' => 'TestPassword_7',
+            'confirm_password' => 'TestPassword_A',
+            'email' => 'testemail7@test.mail',
+            'name' => 'TestName7',
+            'surname' => 'TestSurname7'
+        ];
+        $this->registerPostRequest($data);
+
+        $this->assertEquals(400, $this->client->getResponse()->getStatusCode());
+
+        $expectedErrorCode = '478618a7-95ba-473d-9101-cabd45e49115';
+
+        $this->assertEquals('confirm_password', $this->getPropertyPath());
+        $this->assertEquals($expectedErrorCode, $this->getErrorCode());
+    }
+
+    public function testUnsuccessfulRegisterInvalidPassword(): void
+    {
+        $data = [
+            'username' => 'TestUsername8',
+            'password' => 'TestPassword8',
+            'confirm_password' => 'TestPassword8',
+            'email' => 'testemail8@test.mail',
+            'name' => 'TestName8',
+            'surname' => 'TestSurname8'
+        ];
+        $this->registerPostRequest($data);
+
+        $this->assertEquals(400, $this->client->getResponse()->getStatusCode());
+
+        $expectedErrorCode = 'de1e3db3-5ed4-4941-aae4-59f3667cc3a3';
+
+        $this->assertEquals('password', $this->getPropertyPath());
+        $this->assertEquals($expectedErrorCode, $this->getErrorCode());
+
+        $data['password'] = 'Test_8';
```

Comment:
Read about Data Providers https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html